### PR TITLE
Update diag_table for global coupled iau test for mom6 history output 

### LINKED ISSUE #2089: Update diag_table for global IAU test
## Description
The GFSv17 iau test cpld_control_gfsv17_iau needs fms 2023.02.01 update to correctly output MOM6 averaged fields in IAU configurations. Now with fms 2023.02.01 in ufs WM, we will update diag_able in cpld_control_gfsv17_iau, new outputs are added to data comparison list.

### PR DESCRIPTION
**Pull request by junwang-noaa**
## Commit Queue Requirements:

- [x] Fill out all sections of this template.
- [ ] All sub component pull requests have been reviewed by their code managers.
- [ ] Run the full RT suite (compared to current baselines) on either Hera/Derecho/Hercules AND have committed the log to my PR branch.
- [ ] Add list of all failed regression tests in "Regression Tests" section.

### Description

Update diag_able in cpld_control_gfsv17_iau to output MOM6 history file in correct time.

This PR can be combined with other PRs that change results.

### Commit Message

Update diag_able for global coupled iau test cpld_control_gfsv17_iau for mom6 history output 

### Priority
- [ ] Critical Bugfix (This PR contains a critical bug fix and should be prioritized.)
- [x] High (This PR contains a feature or fix needed for a time-sensitive project (eg, retrospectives, implementations))
- [ ] Normal

### Blocking Dependencies

No.

### Git Issues Fixed By This PR

Closes #2089

### Subcomponent (with links)

- [ ] AQM
- [ ] CDEPS
- [ ] CICE
- [ ] CMEPS
- [ ] CMakeModules
- [ ] FV3
- [ ] GOCART
- [ ] HYCOM
- [ ] MOM6
- [ ] NOAHMP
- [ ] WW3
- [ ] stochastic_physics
- [x] none

### Input data
- [x] No changes are expected to input data.
- [ ] Changes are expected to input data:
  - [ ] New input data.
  - [ ] Updated input data.

### Regression Tests:
- [ ] No changes are expected to any regression test.
- [ x] Changes are expected to the following tests: cpld_control_gfsv17_iau
<details><summary>FAILED REGRESSION TESTS</summary>

</details>

### Libraries

- [x] Not Needed
- [ ] Needed
  - [ ] Create separate issue in [JCSDA/spack-stack](https://github.com/JCSDA/spack-stack) asking for update to library. Include library name, library version.
  - [ ] Add issue link from JCSDA/spack-stack following this item 

### Testing Log:
- RDHPCS
  - [x] Hera
  - [x] Orion
  - [x] Hercules
  - [x] Jet
  - [ ] Gaea
  - [x] Derecho
- WCOSS2
  - [x] Dogwood/Cactus
  - [x] Acorn
- CI
  - [x] Completed
- opnReqTest
  - [ ] N/A
  - [x] Log attached to comment

### CODE REVIEW

**Comment by DeniseWorthen** [comment]
@junwang-noaa Should we try to also fix https://github.com/ufs-community/ufs-weather-model/issues/2084 and update all the diag_tables to at least let G-W have MOM6 output in a separate directory? I'm not sure how to resolve the ATM output directory.

File: tests/parm/diag_table/diag_table_cpld_template [hunk 1]
```diff
@@ -4,65 +4,65 @@
...
- "ocean_model", "geolon_v",    "geolon_v",    "ocn%4yr%2mo%2dy%2hr", "all", .false., "none", 2
- "ocean_model", "geolat_v",    "geolat_v",    "ocn%4yr%2mo%2dy%2hr", "all", .false., "none", 2
-# "ocean_model", "depth_ocean", "depth_ocean", "ocn%4yr%2mo%2dy%2hr", "all", .false., "none", 2
-# "ocean_model", "wet",         "wet",         "ocn%4yr%2mo%2dy%2hr", "all", .false., "none", 2
- "ocean_model", "wet_c",       "wet_c",       "ocn%4yr%2mo%2dy%2hr", "all", .false., "none", 2
- "ocean_model", "wet_u",       "wet_u",       "ocn%4yr%2mo%2dy%2hr", "all", .false., "none", 2
- "ocean_model", "wet_v",       "wet_v",       "ocn%4yr%2mo%2dy%2hr", "all", .false., "none", 2
- "ocean_model", "sin_rot",     "sin_rot",     "ocn%4yr%2mo%2dy%2hr", "all", .false., "none", 2
- "ocean_model", "cos_rot",     "cos_rot",     "ocn%4yr%2mo%2dy%2hr", "all", .false., "none", 2
+ "ocean_model", "geolon",      "geolon",      "./MOM6_OUTPUT/ocn%4yr%2mo%2dy%2hr", "all", .false., "none", 2
```
Comment (DeniseWorthen) [inline tests/parm/diag_table/diag_table_cpld_template]: I think we want this to be parsable, so @[MOM6_OUTPUT]

And should we do the other diag_tables as well (the ones w/ ocean output).

Comment (junwang-noaa) [reply tests/parm/diag_table/diag_table_cpld_template]: @DeniseWorthen you mean @[MOM6_OUTPUT_DIR]?  MOM6_OUTPUT_DIR is defined with value "./MOM6_OUTPUT" in default_vars. I will update

Comment (DeniseWorthen) [reply tests/parm/diag_table/diag_table_cpld_template]: Yes---that is what I meant.

Comment (junwang-noaa) [reply tests/parm/diag_table/diag_table_cpld_template]: Updated.

**Comment by junwang-noaa** [comment]
Pre-test results are on Hera at: /scratch1/NCEPDEV/nems/Jun.Wang/nems/vlab/20240108/ufs-weather-model/tests/logs/RegressionTests_hera.log
All 236 tests passed. the  cpld_control_gfsv17_iau_intel failed because of missing baseline.

FAILED TESTS:
003 cpld_control_gfsv17_iau_intel failed in check_result
cpld_control_gfsv17_iau_intel 003 failed in run_test

**Comment by DeniseWorthen** [comment]
@junwang-noaa Do we really want to move the output location for MOM6 in the RT? I thought the idea is that w/ the MOM6_OUPUT_DIR variable defined, G-W can use MOM6_OUPUT but RTs can continue to write directly to the run directory.

**Comment by junwang-noaa** [comment]
I think it is good to have consistent output structure between RT and G-W.

**Comment by DeniseWorthen** [comment]
But we're not keeping consistent w/rt the RESTART directories or the other component output directories. We specifically made this a variable which could be configured for the RTs independently of where G-W prefers to have their output, restarts etc.

**Comment by BrianCurtis-NOAA** [comment]
Is it possible/worth it to have components output to their own directory and keep it consistent in the restarts?

**Comment by DeniseWorthen** [comment]
I don't understand the advantage for the RT to do that. As I said, we made them variables so that each system can specify them the way that works best of them. The RT  is not the G-W.

**Comment by junwang-noaa** [comment]
The goal is to help developers integrate their updates to UFS WM  and G-W easily, having same values in RT and GW helps us to do that . There is some offline discussion. @DeniseWorthen please let me know if you still have any concerns.

**Comment by DeniseWorthen** [comment]
@junwang-noaa Why are we not making this change in the remaining diag_tables that contain MOM6 output fields? I see OCN output in ``diag_table_p8_gocart_micro`` and ``diag_table_p8_template`` and ``diag_table_template``. If those are not being used anymore then perhaps we should remove them.

I also wonder whether we should take the chance to rename these ``template`` files to ``.IN`` files (as was done for ice_in etc). So ``diag_table_cpld_template_iau`` would become ``diag_table_cpld_iau.IN``

**Comment by junwang-noaa** [comment]
@DeniseWorthen Updated, please review.

File: tests/default_vars.sh [hunk 2]
```diff
@@ -782,6 +784,7 @@ export SYEAR=2021
 export SMONTH=03
 export SDAY=22
 export SHOUR=06
+export CHOUR=06
```
Comment (DeniseWorthen) [inline tests/default_vars.sh:787]: Why do we need ``chour`` here and in the diag_tables?

Comment (junwang-noaa) [reply tests/default_vars.sh:787]: Because the diag table starting time is different for IAU test.

Comment (DeniseWorthen) [reply tests/default_vars.sh:787]: I see, it was already present in the test, which is why I didn't see the assigment. 

What is the "c" in "chour" supposed to mean? For shour, that is "starting hour". So is chour the "continue hour"?

Comment (junwang-noaa) [reply tests/default_vars.sh:787]: Cycle hour. In IAU case, the cycle time is 2021032212, even though the model starts with 2021032206 restart files, the forecast is for 2021032212.

**Comment by junwang-noaa** [comment]
@jkbk2004 Please let me know if PR#2107 will be combined with this one.

**Comment by zach1221** [comment]
> @jkbk2004 Please let me know if PR#2107 will be combined with this one.

I think we do want to combine, I've just requested the branch be synced up on 2107.

**Comment by zach1221** [comment]
@junwang-noaa 2107 is going to take some additional time, due to review of the ccpp-physics sub-pr. If you can sync up your branch here, then we'll just do this PR by itself.

**Comment by junwang-noaa** [comment]
@zach1221 My branch is updated.

**Comment by zach1221** [comment]
Apologies for the delay, testing is complete. Sending out final reviews.

**Review by BrianCurtis-NOAA — APPROVED** [review]

**Review by jkbk2004 — APPROVED** [review]

**Pull request merged**